### PR TITLE
Android64 build fixes

### PR DESCRIPTION
**Pull request by akallabeth**
* Patch for OpenSSL Configure file for android64 architectures
* Fix for broken pthread function detection

Stuff to consider:
* Currently does not work with cmake-2.8 (Thread detection in cmake is broken)
* Currently does not work with android NDK 11 and above (OpenH264 1.5.0 build system does not support that)

### CODE REVIEW

File: scripts/android-build.conf [hunk 1]
```diff
@@ -8,16 +8,16 @@ WITH_JPEG=0
 WITH_OPENH264=1
 WITH_OPENSSL=1
 BUILD_DEPS=1
-ANDROID_NATIVE_API_LEVEL=android-12
+ANDROID_NATIVE_API_LEVEL=android-21
```
Comment (realjiangms) [inline scripts/android-build.conf]: Why we require android-21? Is it required by openssl1.0.2h?

**Comment by akallabeth** [comment]
@realjiangms android 64 requires that, and the config is only meant to be to allow quick starting development. if you want to support older api for 32bit builds just use a config appropriate for the arch

**Comment by realjiangms** [comment]
That makes sense.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by bmiklautz** [comment]
@akallabeth I'll prepare a new machine with recent ndk/sdk for android that this can be tested.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by nfedera** [comment]
lgtm

**Pull request merged**